From 3570ff6c8d148a478acef5c0b3f093a8980c6402 Mon Sep 17 00:00:00 2001
From: "Jose M. Cantera" <jmcanterafonseca@gmail.com>
Date: Mon, 21 Mar 2016 10:21:43 +0100
Subject: [PATCH] Fix CMake to be more generic wrt Ubuntu builds

---
 src/app/contextBroker/CMakeLists.txt | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/src/app/contextBroker/CMakeLists.txt b/src/app/contextBroker/CMakeLists.txt
index 0d3e713..e7f488f 100644
--- a/src/app/contextBroker/CMakeLists.txt
+++ b/src/app/contextBroker/CMakeLists.txt
@@ -75,14 +75,8 @@ link_directories("/opt/lib/")
 
 ADD_EXECUTABLE(contextBroker ${SOURCES} ${HEADERS})
 
-IF ((${DISTRO} STREQUAL "Ubuntu_13.04") OR (${DISTRO} STREQUAL "Ubuntu_12.04.1_LTS") OR (${DISTRO} STREQUAL "Ubuntu_12.04.2_LTS"))
-  MESSAGE("contextBroker: Ubuntu DISTRO: '${DISTRO}'")
-  TARGET_LINK_LIBRARIES(contextBroker ${STATIC_LIBS} -lmongoclient -lboost_thread-mt -lboost_filesystem-mt -lboost_system-mt -lboost_regex-mt -lpthread -lssl -lcrypto -lgnutls -lgcrypt)
-ELSEIF((${DISTRO} STREQUAL "Ubuntu_13.10") OR (${DISTRO} STREQUAL "Ubuntu_14.04.1_LTS") OR (${DISTRO} STREQUAL "Ubuntu_14.10"))
-  MESSAGE("contextBroker: Ubuntu DISTRO: '${DISTRO}'")
-  TARGET_LINK_LIBRARIES(contextBroker ${STATIC_LIBS} -lmongoclient -lboost_thread -lboost_filesystem -lboost_system -lpthread -lssl -lcrypto -lgnutls -lgcrypt)
-ELSEIF((${DISTRO} STREQUAL "Ubuntu_14.04.2_LTS") OR (${DISTRO} STREQUAL "Ubuntu_14.04.3_LTS") OR (${DISTRO} MATCHES "Linux_Mint_.*"))
-  MESSAGE("contextBroker: Ubuntu DISTRO: '${DISTRO}'")
+IF ((${DISTRO} MATCHES "Ubuntu_.*") OR (${DISTRO} MATCHES "Linux_Mint_.*"))
+  MESSAGE("contextBroker: DISTRO: '${DISTRO}'")
   TARGET_LINK_LIBRARIES(contextBroker ${STATIC_LIBS} -lmongoclient -lboost_thread -lboost_filesystem -lboost_system -lboost_regex -lpthread -lssl -lcrypto -lgnutls -lgcrypt)
 ELSEIF(${DISTRO} STREQUAL "openSUSE_13.1")
   MESSAGE("contextBroker: openSUSE DISTRO: '${DISTRO}'")
